pkg/templates/cluster_run_template.go

```diff
@@ -26,91 +26,68 @@ import (
 	"github.com/vmware-tanzu/cartographer/pkg/eval"
 )
 
+func NewRunTemplateModel(template *v1alpha1.ClusterRunTemplate) ClusterRunTemplate {
+	return &runTemplate{
+		template:  template,
+		evaluator: eval.EvaluatorBuilder(),
+	}
+}
+
 type Outputs map[string]apiextensionsv1.JSON
 
 type ClusterRunTemplate interface {
 	GetName() string
 	GetResourceTemplate() v1alpha1.TemplateSpec
-	GetOutput(stampedObjects []*unstructured.Unstructured) (Outputs, *unstructured.Unstructured, error)
+	GetLatestSuccessfulOutput(stampedObjects []*unstructured.Unstructured) (Outputs, *unstructured.Unstructured, error)
 }
 
 type runTemplate struct {
-	template *v1alpha1.ClusterRunTemplate
+	template  *v1alpha1.ClusterRunTemplate
+	evaluator eval.Evaluator
 }
 
-func (t runTemplate) GetOutput(stampedObjects []*unstructured.Unstructured) (Outputs, *unstructured.Unstructured, error) {
-	var (
-		updateError                        error
-		everyObjectErrored                 bool
-		mostRecentlySubmittedSuccesfulTime *time.Time
-		evaluatedStampedObject             *unstructured.Unstructured
-	)
+const StatusPath = `status.conditions[?(@.type=="Succeeded")].status`
```

Comment:
wdyt?

```suggestion
const StatusSucceededPath = `status.conditions[?(@.type=="Succeeded")].status`
```

(not a strong opinion at all)

Reply:
did something similar.